src/odemis/acq/move.py

```diff
@@ -614,6 +618,278 @@ def _doCryoSwitchSamplePosition(self, future, target):
+        dep_val[self._axes_dep["x"]] = map_dep_pos["x"]
+        dep_val[self._axes_dep["y"]] = map_dep_pos["y"]
+
+        return dep_val
+
+    def _updateAxesRange(self):
+        """
+        Calculate the axes range of position VA from the given range of dep's.
+        """
+        # Calculate rectangular border points on the given range
+        min_x, max_x = self.axes["x"].range
+        min_y, max_y = self.axes["y"].range
+        coordinates = [[min_x, min_y],
+                       [min_x, max_y],
+                       [max_x, min_y],
+                       [max_x, max_y]]
+
+        # Calculate range based on the conversion
+        new_coordinates = list(map(self._convertPosFromdep, coordinates))
+
+        # Update default axes range
+        new_range = numpy.array(new_coordinates, dtype=float)
+        self.axes["x"].range = (min(new_range[:, 0]), max(new_range[:, 0]))
+        self.axes["y"].range = (min(new_range[:, 1]), max(new_range[:, 1]))
+
+
+class MeteorTFS2PostureManager(MeteorTFS1PostureManager):
+    def __init__(self, microscope):
+        super().__init__(microscope)
+        # Check required metadata used during switching
+        self.required_keys.add(model.MD_POS_COR)
+        self.check_stage_metadata(required_keys=self.required_keys)
+        if not {"x", "y", "rz", "rx"}.issubset(self.stage.axes):
+            raise KeyError("The stage misses 'x', 'y', 'rx' or 'rz' axes")
+        stage_md = self.stage.getMetadata()
+        self.rotation = stage_md[model.MD_ROTATION_COR]  # pre-tilt of the shuttle
+        # Instantiate a class to convert stage axes such that the imaging plane in the new axes
+        # moves perpendicular to the objective axis by keeping a fixed distance
+        self.convert_stage = ConvertStage(self.stage, ["y", "z"], self.rotation)
+        # Enable the correction in SEM mode when switched from FM mode
+        self.sem_cor = False
+        self.sem_stage_pos_pre = None # previous stage position in SEM before switching to FM
+
+    # Note: this transformation consists of translation of along x and y
+    # axes, and 7 degrees rotation around rx, and 180 degree rotation around rz.
+    # The rotation angles are constant existing in "FM_POS_ACTIVE" metadata,
+    # but the translation are calculated based on the current position and some
+    # correction/shifting parameters existing in metadata "FM_POS_ACTIVE".
+    # This correction parameters can change every session. They are calibrated
+    # at the beginning of each run.
+    def _transformFromSEMToMeteor(self, pos: Dict[str, float]) -> Dict[str, float]:
+        """
+        Transforms the current stage position from the SEM imaging area to the
+        meteor/FM imaging area.
+        :param pos: (dict str->float) the initial stage position.
+        :return: (dict str->float) the transformed position.
+        """
+
+        stage_md = self.stage.getMetadata()
+        self.sem_stage_pos_pre = pos
```

Comment:
setting this here will reset it every time this function is called, and its called frequently to check target position etc for a variety of positions (e.g. grid_1_pos). I don't think thats what we want to do.